### PR TITLE
Cranelift: Harvest each Souper LHS into its own file

### PR DESCRIPTION
**Pull request by fitzgen**
Souper only handles one input LHS at a time, so this makes it way easier to
script. Don't need to try and parse each LHS.

The audits are from when I was originally using `blake3` to hash and dedupe each left-hand side. I gave up on auditing `blake3` when I reached the large amounts of hand-written assembly, but maybe these other audits will be useful to us or someone else one day.

### CODE REVIEW

File: cranelift/src/souper_harvest.rs [hunk 1]
```diff
@@ -77,13 +98,33 @@ pub fn run(options: &Options) -> Result<()> {
...
+            for lhs in recv {
+                let lhs = if keep_harvest_source {
+                    &lhs
+                } else {
+                    // Remove the first `;; Harvested from v12 in u:34` line.
+                    let i = lhs.find('\n').unwrap();
+                    &lhs[i + 1..]
+                };
+                let hash = fxhash::hash(lhs.as_bytes());
+                if already_harvested.insert(hash) {
```
Comment (elliottt) [inline cranelift/src/souper_harvest.rs:115]: Keeping the set of hashes might skip some results due to collisions, right? Are the `lhs` strings too big to store in the HashSet instead?

**Review by elliottt — COMMENTED** [review]
One question about the hash set of hashes, otherwise happy to r+ :)

File: cranelift/src/souper_harvest.rs (hunk 1 above)
Comment (fitzgen) [reply cranelift/src/souper_harvest.rs:115]: Yeah. I just didn't want to keep all the LHSs in memory and blake3 is pretty good an not-yet-known-to-be-broken so it seems safe. Also, I'm pretty sure I'm the only user of this tool :-p

**Review by elliottt — APPROVED** [review]
Looks good to me!

**Pull request merged**